src/webui/www/private/scripts/pathAutofill.js

```diff
@@ -0,0 +1,94 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project"s "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+"use strict";
+
+/*
+File implementing auto-fill for the path input field in the path dialogs.
+*/
+if (window.qBittorrent === undefined)
+    window.qBittorrent = {};
+
+window.qBittorrent.pathAutofill = (() => {
+
+    const exports = () => {
+        return {
+            attachPathAutofill: attachPathAutofill
+        };
+    };
+
+    function showInputSuggestions(inputElement, names) {
+        const datalist = document.createElement("datalist");
+        datalist.id = inputElement.id + "Suggestions";
+        for (const name of names) {
+            const option = document.createElement("option");
+            option.value = name;
+            datalist.appendChild(option);
+        }
+
+        const oldDatalist = document.getElementById(inputElement.id + "Suggestions");
+        if (oldDatalist !== null)
+            oldDatalist.remove();
+
+        inputElement.setAttribute("list", datalist.id);
+        inputElement.appendChild(datalist);
+    }
+
+    async function showPathSuggestions(element, mode) {
+        const partialPath = element.value;
+        if (partialPath === "") {
+            return;
+        }
+        fetch("api/v2/app/getDirectoryContent?dirPath=" + partialPath + "&mode=" + mode)
+            .then(response => response.json())
+            .then(filesList => { showInputSuggestions(element, filesList); });
+    }
+
+    // get all input text fields with class pathDirectory class="pathDirectory"
+    function attachPathAutofill() {
+        async function handleDirSuggestions() {
+            showPathSuggestions(this, "dirs");
+        }
+        async function handleFilesSuggestions() {
+            showPathSuggestions(this, "all");
+        }
+
+        const directoryInputs = document.getElementsByClassName("pathDirectory");
+        for (const input of directoryInputs)
+            input.addEventListener("input", handleDirSuggestions);
+        const fileInputs = document.getElementsByClassName("pathFile");
+        for (let i = 0; i < fileInputs.length; i++) {
+            //listen for typing events
+            fileInputs[i].addEventListener("input", handleFilesSuggestions);
+        }
```

Comment:
```suggestion
        const directoryInputs = document.getElementsByClassName("pathDirectory");
        for (const input of directoryInputs)
            input.addEventListener("input", function() { showPathSuggestions(this, "dirs"); });

        const fileInputs = document.getElementsByClassName("pathFile");
        for (const input of fileInputs)
            input.addEventListener("input", function() { showPathSuggestions(this, "all"); });
```
1. By inline I mean really inlining it.
2. The second loop has the same pattern so I would expect it is changed too.